sale_timesheet_order_line_no_update/models/project.py

```diff
@@ -45,4 +45,4 @@ def _compute_sale_line_id_domain(self):
 
     @api.onchange("sale_line_id")
     def _onchange_sale_line_id(self):
-        self.task_ids.write({"new_sale_line_id": self.sale_line_id})
+        self.task_ids._origin.new_sale_line_id = self.sale_line_id
```

Comment:
I'm thinking that the whole thing is misplaced: you can't update a field on another model in an onchange. That's not the goal of onchanges, and this will cause undesired changes when you discard things. You can:

- Overwrite create/write on project, and do this related write.
- Convert `new_sale_line_id` into a computed writable, and react through the depends.